Metalama.Framework.Engine/Templating/TemplatingDiagnosticDescriptors.cs

```diff
@@ -588,5 +588,13 @@ internal static readonly
                 "The attribute '{0}' is not allowed on the compile-time declaration '{1}', because it wouldn't have the expected effect.",
                 _category,
                 Error );
+
+        internal static readonly DiagnosticDefinition<(IMemberOrNamedType Template, IDeclaration? Target, string RequiredCSharpVersion, string TargetCSharpVersion)> TemplateRequiresHigherCSharpVersion
+            = new(
+                "LAMA0282",
+                "Template requires higher C# version than what is allowed in the project.",
+                "The template '{0}' may not be applicable to '{1}', because it uses C# {2}, while the project uses C# {3}. Consider specifying <LangVersion>{2}</LangVersion> in this project, or removing newer language features from the template and then specifying <MetalamaTemplateLanguageVersion> in the aspect project.",
```

Comment:
There are conceptual issues with this error message:
* Reporting the error for every single target is excessive. It should be reported once per project.
* The template identity is an implementation detail of the aspect. Only the aspect name should be mentioned.
 
To be discussed?